assets/src/settings-page/analytics.js

```diff
@@ -0,0 +1,307 @@
+}
+
+/**
+ * Component for a single analytics entry.
+ *
+ * @param {Object} props Component props.
+ * @param {string} props.entryId Unique ID for the entry.
+ * @param {Function} props.onChange Callback to run when data changes.
+ * @param {Function} props.onDelete Callback to run when the entry is to be DocumentAndElementEventHandlers.
+ * @param {string} props.type The entry type.
+ * @param {string} props.config The config JSON string.
+ */
+function AnalyticsEntry( { entryId = '', onChange, onDelete, type = '', config = '{}' } ) {
+	const isExistingEntry = 0 !== entryId.indexOf( NEW_ENTRY_KEY_PREFIX );
+
+	const analyticsTitle = useMemo( () => {
+		if ( isExistingEntry ) {
+			const newEntrySlug = sprintf(
+				'%1$s%2$s',
+				type ? type + '-' : '',
+				entryId.substr( entryId.length - 6 ),
+			);
+
+			/* translators: %s: the entry slug. */
+			return sprintf( __( 'Analytics: %s', 'amp' ), newEntrySlug );
+		}
+
+		return __( 'Add new entry:', 'amp' );
+	}, [ entryId, isExistingEntry, type ] );
+
+	/**
+	 * Track the validity of the config JSON object. A nonempty custom validity string will block form submission.
+	 */
+	const textAreaRef = useRef();
+	useEffect( () => {
+		if ( ! textAreaRef?.current ) {
+			return;
+		}
+
+		if ( ! config ) {
+			textAreaRef.current.setCustomValidity( '' );
+			return;
+		}
+		try {
+			const parsedValue = JSON.parse( config );
+			if ( null === parsedValue || typeof parsedValue !== 'object' || Array.isArray( parsedValue ) ) {
+				textAreaRef.current.setCustomValidity( __( 'A JSON object is required, e.g. {…}', 'amp' ) );
+			} else {
+				textAreaRef.current.setCustomValidity( '' );
+			}
+		} catch ( e ) {
+			textAreaRef.current.setCustomValidity( e.message );
+		}
+	}, [ config ] );
+
+	/**
+	 * On creation, focus the cursor in the first input in the entry.
+	 */
+	const inputWrapper = useRef();
+	useEffect( () => {
+		if ( ! inputWrapper?.current ) {
+			return;
+		}
+
+		const firstInput = inputWrapper.current.querySelector( 'input' );
+		if ( firstInput ) {
+			firstInput.focus();
+		}
+	}, [] );
+
+	return (
+		<PanelRow className="amp-analytics-entry">
+			<h4>
+				{ analyticsTitle }
+			</h4>
+			<div className="amp-analytics-entry__options" ref={ inputWrapper } id={ `amp-analytics-entry${ entryId }` }>
+				<div className="amp-analytics-entry__text-inputs">
+					<TextControl
+						className="option-input"
+						label={ __( 'Type:', 'amp' ) }
+						onChange={ ( newType ) => {
+							onChange( { type: newType } );
+						} }
+						type="text"
+						required
+						placeholder={ __( 'e.g. googleanalytics', 'amp' ) }
+						value={ type }
+					/>
+
+					<TextControl
+						label={ __( 'ID:', 'amp' ) }
+						type="text"
+						value={ isExistingEntry ? entryId : '' }
+						readOnly
+					/>
+				</div>
+
+				<BaseControl
+					id={ `analytics-textarea-control-${ entryId }` }
+					label={ __( 'JSON Configuration:', 'amp' ) }
+				>
+					<textarea
+						rows="10"
```

Comment:
I entertained this possibility, but for the fields in this section we're using built-in browser form validation, which at this point will do a better job of directing the user to the invalid fields.

I've thought about creating a validation context provider that could collect issues with the form along with their associated elements, and scroll the user to form issues when they try to submit. This would allow us to implement form validation behavior for things like the the mode selection options, which are not standard form elements. I might create an issue for that.